frontend_app_file/src/container/PopupCreateFile.jsx

```diff
@@ -137,68 +182,125 @@ class PopupCreateFile extends React.Component {
+
+    this.setState({ uploadStarted: true })
+
+    state.uploadFileList.forEach((file, index) => this.postFile(file, index))
+  }
+
+  handleDeleteFile = (file) => {
+    const { state } = this
+
+    const uploadFileWithoutDeletedFileList = state.uploadFileList.filter(f => f !== file)
+    switch (uploadFileWithoutDeletedFileList.length) {
+      case 0:
+        this.setState({ uploadFilePreview: false }); break
+      case 1:
+        this.loadUploadFilePreview(uploadFileWithoutDeletedFileList[0]); break
+    }
+
+    this.setState({ uploadFileList: uploadFileWithoutDeletedFileList })
+  }
+
+  getPercentUpload () {
+    const { state } = this
+
+    return Math.round(state.uploadFileList.reduce((accumulator, currentValue) => accumulator + currentValue.percent, 0))
+  }
+
+  isValidateButtonDisabled () {
+    const { state } = this
+
+    if (state.uploadFileList.length === 0 || state.uploadStarted) return true
+    return state.uploadFileList.some(f => f.errorMessage)
+  }
+
+  loadUploadFilePreview = (file) => {
+    if (file.type.includes('image') && file.size <= 2000000) {
+      let reader = new FileReader()
+      reader.onload = e => {
+        this.setState({ uploadFilePreview: e.total > 0 ? e.target.result : false })
+        const img = new Image()
+        img.src = e.target.result
+        img.onerror = () => this.setState({ uploadFilePreview: false })
+      }
+      reader.readAsDataURL(file)
+    }
+  }
+
   render () {
     const { props, state } = this
 
     return (
       <CardPopupCreateContent
-        onClose={this.handleClose}
+        onClose={() => this.handleClose(true)}
         onValidate={this.handleValidate}
         label={props.t(state.config.creationLabel)}
         customColor={state.config.hexcolor}
         faIcon={state.config.faIcon}
-        contentName={state.uploadFile ? 'allowValidate' : ''} // hack to update the "disabled" state of the button
+        contentName={this.isValidateButtonDisabled() ? '' : 'allowValidate'} // hack to update the "disabled" state of the button
         onChangeContentName={() => {}}
         btnValidateLabel={props.t('Validate and create')}
-        customStyle={{ top: 'calc(50% - 177px)' }}
+        customStyle={{ top: '50%', transform: 'translateY(-50%)' }}
       >
         <div>
-          {state.progressUpload.display &&
+          {state.uploadStarted &&
             <PopupProgressUpload
               color={state.config.hexcolor}
-              percent={state.progressUpload.percent}
-              filename={state.uploadFile ? state.uploadFile.name : ''}
+              percent={this.getPercentUpload()}
             />
           }
           <FileDropzone
             onDrop={this.handleChangeFile}
-            onClick={this.handleChangeFile}
             hexcolor={state.config.hexcolor}
             preview={state.uploadFilePreview}
-            filename={state.uploadFile ? state.uploadFile.name : ''}
+            multipleFiles={!state.uploadFilePreview}
```

Comment:
`state.uploadFilePreview` is not supposed to be boolean. Or is it ?
the props `preview` is used as a string in `<FileDropzone />` so the meaning `!state.uploadFilePreview` is unclear